sympy/matrices/expressions/trace.py

```diff
@@ -105,9 +105,10 @@ def arg(self):
     def doit(self, **hints):
         if hints.get('deep', True):
             arg = self.arg.doit(**hints)
-            try:
-                return arg._eval_trace()
-            except (AttributeError, NotImplementedError):
+            result = arg._eval_trace()
+            if result is not None:
+                return result
+            else:
                 return Trace(arg)
```

Comment:
```suggestion
            if (tr :=arg._eval_trace()) is not None:
                return tr
            return Trace(arg)
```

Reply:
I don't think that this suggested change helps readability.